server/services/personService.ts

```diff
@@ -10,15 +10,76 @@ import type {
 } from '../data'
 import { PersonUtils } from '../utils'
 import type { Person } from '@accredited-programmes/models'
+import type { OffenceWithDetail } from '@accredited-programmes/ui'
 import type { Caseload, OffenderSentenceAndOffences } from '@prison-api'
 
+interface OffenceHistory {
+  additionalOffences: Array<OffenceWithDetail>
+  indexOffence?: OffenceWithDetail
+}
+
 export default class PersonService {
   constructor(
     private readonly hmppsAuthClientBuilder: RestClientBuilderWithoutToken<HmppsAuthClient>,
     private readonly prisonApiClientBuilder: RestClientBuilder<PrisonApiClient>,
     private readonly prisonerSearchClientBuilder: RestClientBuilder<PrisonerSearchClient>,
   ) {}
 
+  async getOffenceHistory(token: Express.User['token'], prisonNumber: Person['prisonNumber']): Promise<OffenceHistory> {
```

Comment:
I think we want to use the system token here, rather than the user token (see `getPerson` below), as that has our client credentials permissions on it.